code/game/objects/items.dm

```diff
@@ -181,6 +181,35 @@ GLOBAL_VAR_INIT(embedpocalypse, FALSE) // if true, all items will be able to emb
 
 	var/canMouseDown = FALSE
 
+//modification functionality//
+	var/mods = 0
+
+	//maximum number of modifications an item can have.
+	var/maxmodify = 3
+	var/postfixed = FALSE//prevents more then one postfix
+	//weapon poison effect, injected on every stab if poisoned is TRUE.
+	var/poisoned = FALSE
+	var/poisonpower = 1
+	var/stabpoison = /datum/reagent/toxin
+	//weapon flame effect. Ignites and burns based on burnpower if burner is TRUE
+	var/burner = FALSE
+	var/burnpower = 1
+	//weapon chill effect. Reduces temparture and mildly burns. Applies a slowly-increasing status effect that eventually freezes
+	var/freezer = FALSE
+	var/freezepower = 1
+	//When haunted is true, alt-click the item in question to summon a spirit.
+	var/haunted = FALSE
+	var/possessed = FALSE
+	//builds up bleed, bleedpower dictates
+	var/serrated = FALSE
+	var/bleedpower = 1
+
+//attack speed modification: decreases cooldown between clicks. Lower numbers are faster, higher are slower//
+	var/attackspeed = 1
+
+//causes an effect overlay to appear over your char when the item is held inhand. Most often used for "shielding" effects, see red and blue holy staves for an example.
+	var/overlayed = FALSE
+	var/effect_icon = null//pathing is under icons/effects/effects.dmi
```

Comment:
Just going to say that this is a
# Ton
of vars to be adding to a very very very common type

Reply:
Like, it's a step in the right direction, but there HAS to be a better way to do this (components).